chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -830,3 +870,137 @@ func (b *BitcoindNotifier) RegisterBlockEpochNtfn() (*chainntnfs.BlockEpochEvent
 		}, nil
 	}
 }
+
+// handleMissedBlocks is called when the chain backend misses a series of
+// blocks. It dispatches all relevant notifications for the missed blocks,
+// handling a reorg if necessary.
+func (b *BitcoindNotifier) handleMissedBlocks(newHeight int32) error {
```

Comment:
looks like also can be extracted into a common method that takes the `chainConn`?